DSCResources/MSFT_xDNSServerAddress/MSFT_xDNSServerAddress.psm1

```diff
@@ -55,11 +59,52 @@ function Set-TargetResource
         [String]$InterfaceAlias,
 
         [Parameter(Mandatory)]
-        [ValidateSet("IPv4", "IPv6")]
+        [ValidateSet('IPv4', 'IPv6')]
         [String]$AddressFamily
     )
 
-    Test-Properties @PSBoundParameters -Apply
+    try
+    {        
+        Write-Verbose -Message ( @("$($MyInvocation.MyCommand): "
+            'Applying the DNS Server Address ...'
+            ) -join '')
+
+        #Should not need to validate the settings because Test-TargetResource already did this
+
+        #Get the current DNS Server Addresses based on the parameters given.
+        $currentAddress = (Get-DnsClientServerAddress `
+            -InterfaceAlias $InterfaceAlias `
+            -AddressFamily $AddressFamily `
+            -ErrorAction Stop).ServerAddresses
+
+        #Check if the Server addresses are the same as the desired addresses.
+        if(@(Compare-Object `
```

Comment:
Actually on second thought, why do we need this? If Test-TargetResource indicates we need to make a change why are we again seeing if it matches?

In fact I would also argue that we don't even need the try/catch in the set-targetresource. Set-DnsClientServerAddress should have its own erorr messaging that would probably clearer to resolve through research then the error message we're obscuring by throwing our own.  If we were to do that then the entire function would consist of the messaging around begining, and ending and applying the change. This would greatly reduce the logic and testing area on the function.